cpp/daal/src/algorithms/svm/svm_train_thunder_impl.i

```diff
@@ -195,28 +161,144 @@ services::Status SVMTrainImpl<thunder, algorithmFPType, cpu>::compute(const Nume
             DAAL_CHECK_STATUS(status, cachePtr->getRowsBlock(wsIndices, nWS, kernelSOARes));
         }
 
-        DAAL_CHECK_STATUS(status, SMOBlockSolver(y, grad, wsIndices, kernelSOARes, nVectors, nWS, cw, eps, tau, buffer.get(), I.get(), alpha,
-                                                 deltaAlpha.get(), diff));
+        DAAL_CHECK_STATUS(status, SMOBlockSolver(y, grad, wsIndices, kernelSOARes, nVectors, nWS, cw, accuracyThreshold, tau, buffer.get(), I.get(),
+                                                 alpha, deltaAlpha.get(), diff));
 
-        DAAL_CHECK_STATUS(status, updateGrad(kernelSOARes, deltaAlpha.get(), gradBuff.get(), grad, nVectors, nWS));
-        if (checkStopCondition(diff, diffPrev, eps, sameLocalDiff) && iter >= nNoChanges) break;
+        DAAL_CHECK_STATUS(status, updateGrad(kernelSOARes, deltaAlpha.get(), gradBuff.get(), grad, nVectors, nTrainVectors, nWS));
+        if (checkStopCondition(diff, diffPrev, accuracyThreshold, sameLocalDiff) && iter >= nNoChanges) break;
         diffPrev = diff;
     }
 
+    if (svmType == SvmType::regression)
+    {
+        DAAL_ASSERT(alphaTArray.size() == 2 * nVectors);
+        PRAGMA_IVDEP
+        PRAGMA_VECTOR_ALWAYS
+        for (size_t i = 0; i < nVectors; ++i)
+        {
+            alpha[i] = alpha[i] - alpha[i + nVectors];
+        }
+    }
+
     cachePtr->clear();
     SaveResultTask<algorithmFPType, cpu> saveResult(nVectors, y, alpha, grad, cachePtr.get());
     DAAL_CHECK_STATUS(status, saveResult.compute(*xTable, *static_cast<Model *>(r), cw));
 
     return status;
 }
 
+template <typename algorithmFPType, CpuType cpu>
+services::Status SVMTrainImpl<thunder, algorithmFPType, cpu>::classificationInit(NumericTable & yTable, const NumericTablePtr & wTable,
+                                                                                 const algorithmFPType C, algorithmFPType * y, algorithmFPType * grad,
```

Comment:
const is inconsistent here. 'C' violates guidelines (case).

Reply:
Done